packages/react-native-permissions/src/withReactNativePermissions.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  withDangerousMod,
+  ConfigPlugin,
+  createRunOncePlugin,
+} from "@expo/config-plugins";
+import { mergeContents } from "@expo/config-plugins/build/utils/generateCode";
+import fs from "fs";
+import path from "path";
+
+const withReactNativePermissions: ConfigPlugin<{ pods: string[] }> = (
+  config,
+  { pods }
+) => {
+  return withDangerousMod(config, [
+    "ios",
+    async (config) => {
+      const filePath = path.join(
+        config.modRequest.platformProjectRoot,
+        "Podfile"
+      );
+
+      const contents = fs.readFileSync(filePath, "utf-8");
+
+      const result = mergeContents({
+        tag: "react-native-permissions",
+        src: contents,
+        newSrc: getPodfileContent(pods),
+        anchor: /use_native_modules/,
+        offset: 0,
+        comment: "#",
+      });
+
+      fs.writeFileSync(filePath, result.contents);
+
+      return config;
+    },
+  ]);
+};
+
+function getPodfileContent(pods: string[]) {
+  return `
+  permissions_path = '../node_modules/react-native-permissions/ios'
```

Comment:
This should use Node module resolution to support monorepos, [ex](https://github.com/expo/config-plugins/blob/ed020d236b8fa4a810adf90a9773d9a2fba36fd4/packages/ffmpeg-kit-react-native/src/withCocoaPodsImport.ts#L40).

Basically you use Node.js to search for `react-native-permissions/package.json` then you get a string like `../node_modules/react-native-permissions/package.json` which you can then append `../ios` to. This will allow the module to live anywhere and keep the package working in most cases.